pkg/autoops/api/progressive_rollout_operation.go

```diff
@@ -18,35 +18,36 @@ import (
 	"github.com/golang/protobuf/ptypes"
 
 	"github.com/bucketeer-io/bucketeer/v2/pkg/autoops/domain"
-	err "github.com/bucketeer-io/bucketeer/v2/pkg/error"
 	ftdomain "github.com/bucketeer-io/bucketeer/v2/pkg/feature/domain"
 	autoopsproto "github.com/bucketeer-io/bucketeer/v2/proto/autoops"
 	featureproto "github.com/bucketeer-io/bucketeer/v2/proto/feature"
 )
 
-var errVariationNotFound = err.NewErrorNotFound(
-	err.AutoopsPackageName,
-	"a variation for a progressive rollout is not found",
-	"variation_id",
-)
-
 const totalVariationWeight = int32(100000)
 
 func ExecuteProgressiveRolloutOperation(
 	progressiveRollout *domain.ProgressiveRollout,
 	feature *ftdomain.Feature,
 	scheduleID string,
 ) (*featureproto.Strategy, error) {
-	var variationID string
+	// Extract control and target variation IDs
+	controlVariationID, err := progressiveRollout.GetControlVariationID()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
While the `progressiveRollout.GetTargetVariationID` has backward compatibility for existing progressive rollouts, the control `GetControlVariationID` will return empty, causing an invalid state [here](https://github.com/bucketeer-io/bucketeer/pull/2199/files#diff-e646b0b027621b53e88cf426f0b2cacae751c7c353bd298f22e9b735287d5f0fR114).

For existing progressive rollouts, you must set the control variation in the `GetControlVariationID`. It should be the non-target variation (We only allow 2 for existing rollouts).

Please add unit tests to ensure it works as expected.